web/src/stream_popover.ts

```diff
@@ -447,7 +447,9 @@ export async function build_move_topic_to_stream_popover(
         // not changed.
         util.the($<HTMLButtonElement>("#move_topic_modal button.dialog_submit_button")).disabled =
             Number.parseInt(current_stream_id, 10) === select_stream_id &&
-            (new_topic_name === undefined || new_topic_name.trim() === old_topic_name.trim());
+            (new_topic_name === undefined ||
+                new_topic_name.trim() === old_topic_name.trim() ||
+                !new_topic_name.trim());
```

Comment:
As discussed on CZO, the button should be disabled only when `manadatory_topics` is true.

Reply:
Also, I guess the button should be disabled when `mandatory_topics` is true and the topic input is empty, irrespective of whether the stream is changed or not.